sklearn/gaussian_process/kernels.py

```diff
@@ -355,6 +355,10 @@ def diag(self, X):
     def is_stationary(self):
         """Returns whether the kernel is stationary. """
 
+    @abstractmethod
+    def vector_X_only(self):
```

Comment:
I'm not sure about this naming "requires_vector_X" is a bit better to me, but even then I find it awkward as a method and suspect it should be a scalar attribute or property. Don't entirely mind it as a method though

Reply:
I was just following paradigm of `StationaryKernelMixin` and `is_stationary()`, but I do agree the name is not the most descriptive so far. What about `requires_vector_input` or `vector_input_only`?

Reply:
`requires_vector_input` would be fine in the manner of `is_stationary` (both using present-tense verbs)

Reply:
Done.